plugin/group_replication/src/certifier.cc

```diff
@@ -1223,8 +1247,16 @@ void Certifier::garbage_collect() {
   }
 #endif
 
-  mysql_mutex_unlock(&LOCK_certification_info);
+    mysql_mutex_unlock(&LOCK_certification_info);
+    if (it == certification_info.end()) {
+      break;
+    }
+    /* could add else clause with a short sleep */
+    my_sleep(get_certification_loop_sleep_time_var());
```

Comment:
Based on the comment from Venkatesh, I just realized another unlikely, but possible issue:

We currently:
* get to a point with this iterator, based on m_certification_info
* release the lock
* wait some time
* lock again
* continue work with the same iterator, assuming that it is still valid

But are we 100% sure that the iterator doesn't get invalidated while we wait? We don't hold the lock, any other method can change the collection.

Currently it's a map, and only `garbage_collect()` and `clear_certification_info()` delete items from it. But:

1. I'm not sure that with the lock relaxed, these can't happen in parallel
2. Will we notice if upstream changes the type of the container, or adds other methods that remove items?